var/spack/repos/builtin/packages/py-fenics-dolfinx/package.py

```diff
@@ -24,8 +24,10 @@ class PyFenicsDolfinx(PythonPackage):
     depends_on("cmake@3.19:", type="build")
     depends_on("hdf5", type="build")
     depends_on("pkgconfig", type="build")
-    depends_on("python@3.8:", type=("build", "run"), when="@0.5:")
-    depends_on("python@3.7:", type=("build", "run"))
+
+    depends_on("python@3.8:", type=("build", "run"), when="@main")
+    depends_on("python@3.8:3.10", type=("build", "run"), when="@0.5:")
+    depends_on("python@3.7:3.10", type=("build", "run"))
```

Comment:
```suggestion
    depends_on("python@3.8:", when="@0.6.1:", type=("build", "run"))
    depends_on("python@3.8:3.10", when="@0.5:0.6.0", type=("build", "run"))
    depends_on("python@3.7:3.10", when="@0.4", type=("build", "run"))
```
If you know it will be fixed in 0.6.1 you can use this, otherwise you'll have to update it when the next release comes out.

`@main` is included in `@0.5:` so both constraints would have applied.

Reply:
Thanks. Yes, it will be fixed in the next release.